definitions/3.0.0/messagePayloadObject.json

```diff
@@ -0,0 +1,147 @@
+                "application/vnd.aai.asyncapi;version=2.4.0",
+                "application/vnd.aai.asyncapi+json;version=2.4.0",
+                "application/vnd.aai.asyncapi+yaml;version=2.4.0",
+                "application/vnd.aai.asyncapi;version=2.5.0",
+                "application/vnd.aai.asyncapi+json;version=2.5.0",
+                "application/vnd.aai.asyncapi+yaml;version=2.5.0",
+                "application/vnd.aai.asyncapi;version=2.6.0",
+                "application/vnd.aai.asyncapi+json;version=2.6.0",
+                "application/vnd.aai.asyncapi+yaml;version=2.6.0",
+                "application/vnd.aai.asyncapi;version=3.0.0",
+                "application/vnd.aai.asyncapi+json;version=3.0.0",
+                "application/vnd.aai.asyncapi+yaml;version=3.0.0"
+              ]
+            }
+          }
+        },
+        "then": {
+          "properties": {
+            "schema": {
+              "$ref": "http://asyncapi.com/definitions/3.0.0/schema.json"
+            }
+          }
+        }
+      },
+      {
+        "if": {
+          "required": [
+            "schemaFormat"
+          ],
+          "properties": {
+            "schemaFormat": {
+              "enum": [
+                "application/schema+json;version=draft-07",
+                "application/schema+yaml;version=draft-07"
+              ]
+            }
+          }
+        },
+        "then": {
+          "properties": {
+            "schema": {
+              "$ref": "http://json-schema.org/draft-07/schema"
+            }
+          }
+        }
+      },
+      {
+        "if": {
+          "required": [
+            "schemaFormat"
+          ],
+          "properties": {
+            "schemaFormat": {
+              "enum": [
+                "application/vnd.oai.openapi;version=3.0.0", 
+                "application/vnd.oai.openapi+json;version=3.0.0", 
+                "application/vnd.oai.openapi+yaml;version=3.0.0"
+              ]
+            }
+          }
+        },
+        "then": {
+          "properties": {
+            "schema": {
+              "$ref": "http://asyncapi.com/definitions/3.0.0/openapiSchema_3_0.json"
+            }
+          }
+        }
+      },
+      {
+        "if": {
+          "required": [
+            "schemaFormat"
+          ],
+          "properties": {
+            "schemaFormat": {
+              "enum": [
+                "application/vnd.apache.avro;version=1.9.0",
+                "application/vnd.apache.avro+json;version=1.9.0",
+                "application/vnd.apache.avro+yaml;version=1.9.0"
+              ]
+            }
+          }
+        },
+        "then": {
+          "properties": {
+            "schema": {
+              "$ref": "http://asyncapi.com/definitions/3.0.0/avroSchema_v1.json"
+            }
+          }
+        }
+      }
+    ],
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "http://asyncapi.com/definitions/3.0.0/messagePayloadObject.json"
```

Comment:
I would call it something similar to `MultiFormatSchemaObject` that is unrelated to `payload`.